internal/cli/cmd/login/login.go

```diff
@@ -177,13 +182,36 @@ func (o *option) authAndSetToken() error {
 			return ctx.Err()
 		case <-ticker.C:
 			tokenResponse, tokenErr := o.apiClient.Auth().DeviceAccessToken(string(o.externalAuthKind), o.externalAuthProvider, deviceCodeResponse.DeviceCode)
-			if tokenErr != nil {
-				continue
+			if tokenErr == nil {
+				// Handle the access token
+				o.accessToken = tokenResponse.AccessToken
+				o.refreshToken = tokenResponse.RefreshToken
+				return nil
 			}
-			// Handle the access token
-			o.accessToken = tokenResponse.AccessToken
-			o.refreshToken = tokenResponse.RefreshToken
-			return nil
+
+			reqErr := &perseshttp.RequestError{}
+			if errors.As(tokenErr, &reqErr) && reqErr.Err != nil {
+				oauthErr := &api2.OAuthError{}
+				if errors.As(reqErr.Err, &oauthErr) {
+					switch oauthErr.ErrorCode {
+					case errSlowDown:
+						// https://datatracker.ietf.org/doc/html/rfc8628#section-3.5
+						// "the interval MUST be increased by 5 seconds for this and all subsequent requests"
+						interval += 5
+						ticker.Reset(time.Duration(interval) * time.Second)
+						continue
+					case errAuthorizationPending:
+						// Do nothing.
+						continue
+					case errAccessDenied, errExpiredToken:
```

Comment:
maybe it doesn't worth to put this case if we ended to execute the default case, wdyt ?

Reply:
Yes, I wondered the same haha and I don't know why but I copied from golang/oauth2 and they do this. I think you're right I will modify